File: metal_test.m
// Simple Objective-C program to test Metal device creation
#import <Foundation/Foundation.h>
#import <Metal/Metal.h>

int main() {
    @autoreleasepool {
        NSLog(@"=== Objective-C Metal Device Test ===");
        
        // Try to create a Metal device
        id<MTLDevice> device = MTLCreateSystemDefaultDevice();
        
        if (device) {
            NSLog(@"SUCCESS: Metal device created: %@", device.name);
        } else {
            NSLog(@"ERROR: Failed to create Metal device");
        }
    }
    
    return 0;
}

File: metal_test.c
// Simple C program to test Metal device creation
#include <stdio.h>
#include <Metal/Metal.h>

int main() {
    printf("=== C Metal Device Test ===\n\n");
    
    // Try to create a Metal device
    id<MTLDevice> device = MTLCreateSystemDefaultDevice();
    
    if (device) {
        printf("SUCCESS: Metal device created: %s\n", [[device name] UTF8String]);
    } else {
        printf("ERROR: Failed to create Metal device\n");
    }
    
    return 0;
}
